pydeseq2/DeseqDataSet.py

```diff
@@ -181,15 +182,22 @@ def __init__(
             raise ValueError(
                 "The count matrix and clinical data should have the same index."
             )
-        self.design_factor = design_factor
-        if self.clinical[self.design_factor].isna().any():
+
+        # Convert design factor to list if a single string was provided.
+        self.design_factors = (
+            [design_factors] if isinstance(design_factors, str) else design_factors
+        )
+
+        if self.clinical[self.design_factors].isna().any().any():
             raise ValueError("NaNs are not allowed in the design factor.")
-        self.clinical[self.design_factor] = self.clinical[self.design_factor].astype(str)
+        self.clinical[self.design_factors] = self.clinical[self.design_factors].astype(
+            str
+        )
 
-        # Build the design matrix (splits the dataset in two cohorts)
+        # Build the design matrix (splits the dataset in cohorts)
         self.design_matrix = build_design_matrix(
             self.clinical,
-            design_factor,
+            design_factors,
```

Comment:
just a minor comment: I would suggest to add the argument names which adds for the clarity of read (not required for merge though)

Reply:
Hi Maria, not sure what you mean, do you mean we should write `design=design_factors` ?

Reply:
I've named all arguments in [3c0ded6](https://github.com/owkin/PyDESeq2/pull/32/commits/3c0ded613d38aacac433983c9871e3f3210c37b5)

Reply:
yes, that's what I meant. Thanks :-)